docs: add missing punctuation in document

lumirlumir opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [OpenJS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

<!--
    The following template is intentionally not a markdown checkbox list for the reasons
    explained in https://github.com/eslint/eslint/pull/12848#issuecomment-580302888
-->

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/latest/contribute/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

Hello, I've added missing punctuation in `governance.md`.

There is always punctuation at the end of the lists (represented with `*` in Markdown), as shown in the screenshot below. However, the items I added were missing it.

![image](https://github.com/user-attachments/assets/995b0b25-bb6a-4b0f-82cc-2bb71071ca08)

#### Is there anything you'd like reviewers to focus on?

```markdown
* Technical direction
* Project governance and process (including this policy)
* Contribution policy
* GitHub repository hosting
```
The sentences in the list above are incomplete, so it’s possible to omit the periods. However, since periods are used consistently throughout the Markdown page, I’m debating whether to include them here as well.

<!-- markdownlint-disable-file MD004 -->

Tanujkanti4441 reviewed (commented):
Hi @lumirlumir, thanks for the PR!

There is inconsistency regarding `full stop` punctuation in docs pages like:

We have full stop here in list
![Screenshot 2024-11-25 190356](https://github.com/user-attachments/assets/c12d708e-395f-4dc5-81d2-14d213b557d6)

but not here
![Screenshot 2024-11-25 190333](https://github.com/user-attachments/assets/1a4dfad2-2822-4de4-affc-309914c5869c)

And there are few more place where punctuation is not consistent in the same list such as:

in [manage-issues](https://eslint.org/docs/latest/maintain/manage-issues) page.
![Screenshot 2024-11-25 190206](https://github.com/user-attachments/assets/ed6b3596-6d0e-4c0f-82f3-b736a56e60be)

or in [work-on-issue](https://eslint.org/docs/latest/contribute/work-on-issue) page
![Screenshot 2024-11-25 190139](https://github.com/user-attachments/assets/92fb3430-79b7-45f4-9d6b-ffa610b7ba62)

i think we should also make it consistent on other pages.

lumirlumir commented:
Hello, @Tanujkanti4441

Thank you for the detailed review!  

I've actually been checking for other missing punctuation in the documentation. If that's okay, would it be alright to add the missing punctuation to all the other document pages as well? I've noticed that many sentences in the lists are missing punctuation.  

So far, I've reviewed about a third of the documents and found missing punctuation in more than 10 pages. By the time I'm done, I estimate that there could be issues in over 20 pages.  

Now, I'm wondering how best to proceed with this task:  

1. Find all the missing punctuation, integrate them, and update this PR. However, this might make the PR too large to review.  
2. Split the process by category directories (e.g., "INTEGRATE ESLINT," "CONTRIBUTE TO ESLINT," etc.). 
3. ...Other suggestions...

It would be helpful to get your advice on this. I'll follow your recommendation!  

Once the direction is decided, I'll let you know when I expect to finish this work.

Tanujkanti4441 commented:
> Now, I'm wondering how best to proceed with this task:
> 
>     1. Find all the missing punctuation, integrate them, and update this PR. However, this might make the PR too large to review.
> 
>     2. Split the process by category directories (e.g., "INTEGRATE ESLINT," "CONTRIBUTE TO ESLINT," etc.).
> 
>     3. ...Other suggestions...

IMHO, it's better to update this PR because on the basis of what i have noticed in docs i am assuming even the number of pages would be large but the content to update (adding punctuation) won't be that much in per page.

But still let's ask other @eslint/eslint-team members that how they would like the PR to be made.

nzakas commented on docs/src/contribute/governance.md:
```diff
@@ -111,10 +111,10 @@ The ESLint project is jointly governed by a Technical Steering Committee (TSC) w
-* Project governance and process (including this policy)
-* Contribution policy
-* GitHub repository hosting
+* Technical direction.
+* Project governance and process (including this policy).
+* Contribution policy.
+* GitHub repository hosting.
```
These should not have periods.

nzakas commented:
I think it's fine to update this PR will all of the changes.

Just note that not all bullets required periods, only those that are complete sentences.

lumirlumir commented:
Thanks, @Tanujkanti4441 and @nzakas.

I think I can finish this by the end of the week. Until then, I'll keep it as a draft.  

Once the process is complete, I'll ping the reviewers!

lumirlumir commented:
Hello, @Tanujkanti4441 @nzakas,  

I've added as many missing punctuation marks as I could.  

I only added them to complete sentences, not to incomplete ones.  

Additionally, the `Yes.` in the screenshot below uses a `.` while the others don’t. For consistency, I deleted it.

nzakas reviewed (approved):
LGTM. Thanks!
